source/fundamentals/aggregation-expression-operations.txt

```diff
@@ -0,0 +1,1221 @@
+
+   [ { $project: {
+       lowInventory: {
+           $minN: {
+             input: { $objectToArray: "$inventory" },
+             n: 2
+           } }
+   } } ]
+
+String Operations
+~~~~~~~~~~~~~~~~~
+
+You can perform a string operation on a value of type ``MqlString``
+using the methods described in this section.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 50 50
+
+   * - Java Method
+     - Aggregation Pipeline Operator
+
+   * - `append() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#append(com.mongodb.client.model.mql.MqlString)>`__
+     - :manual:`$concat </reference/operator/aggregation/concat/>`
+
+   * - `length() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#length()>`__
+     - :manual:`$strLenCP </reference/operator/aggregation/strLenCP/>`
+
+   * - `lengthBytes() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#lengthBytes()>`__
+     - :manual:`$strLenBytes </reference/operator/aggregation/strLenBytes/>`
+
+   * - `substr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#substr(int,int)>`__
+     - :manual:`$substrCP </reference/operator/aggregation/substrCP/>`
+
+   * - `substrBytes() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#substrBytes(int,int)>`__
+     - :manual:`$substrBytes </reference/operator/aggregation/substrBytes/>`
+
+   * - `toLower() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#toLower()>`__
+     - :manual:`$toLower </reference/operator/aggregation/toLower/>`
+
+   * - `toUpper() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#toUpper()>`__
+     - :manual:`$toUpper </reference/operator/aggregation/toUpper/>`
+
+Suppose you need to generate lowercase usernames for employees of a
+company from the employees' last names and employee IDs.
+
+The ``append()`` method combines the ``firstName`` and ``lastName`` fields into
+a single username, while the ``toLower()`` method makes the entire username
+lowercase. This example stores the result as a new ``username`` field using
+the ``project()`` method.
+
+The following code shows the pipeline for this aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   var lastName = current().getString("lastName");
+   var employeeID = current().getString("employeeID");
+
+   asList(project(fields(
+       computed("username", lastName
+           .append(employeeID)
+           .toLower())
+   )));
+   
+The following code provides an equivalent aggregation pipeline in
+the Query API:
+
+.. code-block:: javascript
+   :copyable: true
+
+   [ { $project: {
+       username: {
+           $toLower: { $concat: ["$lastName", "$employeeID"] } }
+   } } ]
+
+Type-Checking Operations
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can perform a type-check operation on a value of type ``MqlValue``
+using the methods described in this section.
```

Comment:
```suggestion
using the methods described in this section. These methods do not return boolean values, but instead allow the user to specify a default value to use when the original value, of unknown type, is not of the specified type. To perform branching logic based on type checks, see switchOn.
```
I added the (imperfect) wording above because it seems odd to call these type checks. Usually, a type-check is like  `o instanceof String` in Java, which returns a boolean, and allows for branching logic. These "type checks" are more like "casts" with a default value in case the cast fails. I guess they could be called "Casting Operations" but I suspect that would be more confusing.